oak-core/src/test/java/org/apache/jackrabbit/oak/namepath/impl/NamePathMapperImplTest.java

```diff
@@ -135,6 +135,26 @@ public void testOakToJcr() {
         }
     }
 
+    @Test
+    public void testOakToExpandedJcr() {
+        assertEquals("/{http://www.example.com/foo}bar", npMapper.getExpandedJcrPath("/oak-foo:bar"));
+        assertEquals("/{http://www.example.com/foo}bar/{http://www.example.com/quu}qux", npMapper.getExpandedJcrPath("/oak-foo:bar/oak-quu:qux"));
+        assertEquals("{http://www.example.com/foo}bar", npMapper.getExpandedJcrPath("oak-foo:bar"));
+        assertEquals(".", npMapper.getExpandedJcrPath(""));
+
+        try {
+            npMapper.getExpandedJcrPath("{http://www.jcp.org/jcr/nt/1.0}unstructured");
```

Comment:
@reschke the argument is always expected to be an oak name (i.e. always qualified form), everything else is an error condition leading deliberately to an exception. However this can never be triggered when called from `JackrabbitSession.getExpandedName(Item item)` as the item at that point in time needs to have a oak name already.